src/OpenTelemetry.Extensions/Internal/ActivityEventAttachingLogProcessor.cs

```diff
@@ -83,7 +83,11 @@ public override void OnEnd(LogRecord data)
             }
 
             var activityEvent = new ActivityEvent("log", data.Timestamp, tags);
-            activity.AddEvent(activityEvent);
+
+            if (this.options.Filter?.Invoke(activityEvent) != false)
```

Comment:
Typically anytime we call into user code we have a try/catch with an eventsource log for any thrown exceptions. Basically to keep things safe. Example right above if you want to take a stab at adding that.

Reply:
Sure, it's there